jcl/src/java.base/share/classes/java/lang/Access.java

```diff
@@ -441,5 +441,26 @@ public void addExports(Module fromModule, String pkg) {
 	}
 /*[ENDIF] JAVA_SPEC_VERSION >= 16 */
 
+/*[IF JAVA_SPEC_VERSION >= 17]*/
+	// This implementation can be replaced with invoking of String.decodeASCII() after switching to RI String.
+	public int decodeASCII(byte[] srcBytes, int srcPos, char[] dstChars, int dstPos, int length) {
+		int numDecoded = 0;
+		while (numDecoded < length) {
```

Comment:
Looking at callers, this needs to check the src position as well.

Reply:
> Looking at callers, this needs to check the src position as well.

Which caller is it and what check is required for `srcPos`? The callers I found ensure proper `srcPos`/`dstPos`/`length` are supplied to this internal API for optimization.

Reply:
Some of the callers are passing in the min of src length and dest length. This code is only checking against dest.

Reply:
Actually all the callers pass the min, so this is fine.